### PR TITLE
Remove CFG_ from vexpress and various platforms

### PR DESCRIPTION
**Pull request by etienne-lms**
This P-R removes CFG_ definitions from platform_config.h files for the
platforms vexpress, dl02, hikey, marvell, mediatek, poplar, rcar, rpi3, sam,
sprd, zynqmp. 

For vexpress, there is a specific commit linaro can ack. All others aggregated 
linaro can ack the as obvious changes without asking maintainers.

### CODE REVIEW

File: core/arch/arm/plat-hikey/conf.mk [hunk 1]
```diff
@@ -2,6 +2,9 @@ PLATFORM_FLAVOR ?= hikey
 
 include core/arch/arm/cpu/cortex-armv8-0.mk
 
+CFG_TEE_CORE_NB_CORE = 8
+
```
Comment (jenswi-linaro) [inline core/arch/arm/plat-hikey/conf.mk]: Empty line not needed.

File: core/arch/arm/plat-rpi3/conf.mk [hunk 2]
```diff
@@ -1,5 +1,8 @@
 include core/arch/arm/cpu/cortex-armv8-0.mk
 
+CFG_TEE_CORE_NB_CORE = 4
+
```
Comment (jenswi-linaro) [inline core/arch/arm/plat-rpi3/conf.mk]: Empty line not needed.

**Comment by jenswi-linaro** [comment]
`"plat-vexpress: move CFG_TEE_CORE_NB_CORE to platform conf.mk"`
`Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>`

File: core/arch/arm/plat-hikey/conf.mk (hunk 1 above)
Comment (etienne-lms) [reply core/arch/arm/plat-hikey/conf.mk]: addressed. remaining empty line is ok.

File: core/arch/arm/plat-rpi3/conf.mk (hunk 2 above)
Comment (etienne-lms) [reply core/arch/arm/plat-rpi3/conf.mk]: addressed. remaining empty line is ok.

**Comment by etienne-lms** [comment]
tags applied, rebased.

**Comment by jforissier** [comment]
`plat-xxx` does not exist

**Comment by etienne-lms** [comment]
True. I found this header line more easy track related to those for the other platforms.
Would you prefer `move CFG_TEE_CORE_NB_CORE to conf.mk for various platforms`?

**Comment by etienne-lms** [comment]
Rebased and updated 2nd commit header line from
`plat-xxx: move CFG_TEE_CORE_NB_CORE from platform_config to conf.mk`
to
`core: move CFG_TEE_CORE_NB_CORE to conf.mk for various platforms`.

**Pull request merged**